contribute/design/1.x/cstor/cstor-status.md

```diff
@@ -0,0 +1,220 @@
+	// The reason for the condition's last transition.
+	Reason string `json:"reason,omitempty" protobuf:"bytes,4,opt,name=reason"`
+	// A human readable message indicating details about the transition.
+	Message string `json:"message,omitempty" protobuf:"bytes,5,opt,name=message"`
+}
+```
+
+When the conditions like expansion or disk replacement is under progress the message and reason fields will get populated with corresponding details and once the condition has reached completion the fields will have some default values as placeholders until the condition is triggered again.
+
+The proposed conditions for CSPI are :
+
+**PodAvailable** : The PodAvailable condition represents whether the CSPI pool pod is running or not. Whenever the PodAvailable is set to False then the CSPI phase should be set to Unavail to tackle the stale phase on the CSPI when the pool pod is not in running state. The owner of this condition will be the CSPC operator as when the pool pod is lost the cspi-mgmt will not be able to update the conditions.
+```yaml
+Conditions:
+  - lastUpdateTime: 
+    lastTransitionTime: 
+    status: "True"
+    type: PodAvailable
+
+Conditions:
+  - lastUpdateTime: 
+    lastTransitionTime: 
+    message: 'pool pod not in running state'
+    reason: MissingPoolDeployment
+    status: "False"
+    type: PodAvailable
+```
+
+**Created** : The Created condition represents whether the pool got created successfully or not.
+```yaml
+Conditions:
+  - lastUpdateTime: 
+    lastTransitionTime: 
+    status: "True"
+    type: Created
+
+Conditions:
+  - lastUpdateTime: 
+    lastTransitionTime: 
+    message: ‘error msg from zfs command'
+    reason: CreationFailed
+    status: "False"
+    type: Created
+```
+
+**Expanded** : The Expanded condition gets appended when someone triggers pool expansion and represents the status of expansion. If multiple vdev were added for expansion then condition.status will be set as true further information will be available on events of corresponding CSPI.
+```yaml
+Conditions:
+  - lastUpdateTime: 
+    lastTransitionTime: 
+    status: "True"
+    type: Expanded
+
+Conditions:
+  - lastUpdateTime: 
+    lastTransitionTime: 
+    reason: ExpansionInProgress
+    status: "Unknown"
+    type: Expanded
+```
+
+**DiskReplaced** : The DiskReplaced condition gets appended when someone triggers disk replacement on that pool and represents the status of replacement. If multiple disks were replacing then condition message will show that the following are block devices that were under replacement. Further information will be available on corresponding CSPI events.
+```yaml
+Conditions:
+  - lastUpdateTime: 
+    lastTransitionTime: 
+    status: "True"
+    type: DiskReplaced
+
+Conditions:
+  - lastUpdateTime: 
+    lastTransitionTime: 
+    message: ‘error msg from zfs command'
+    reason: ReplacementFailed
+    status: "False"
```

Comment:
what would be the 'status' when few replacements are succeeded and few are NOT?
same question for other Conditions as well..

Reply:
I think we shouldn't say anything related to failures in the conditions it will be continuous operations(keep on trying) but we will update it with a propper message in reason.